server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderator.java

```diff
@@ -465,6 +465,7 @@ public void onFailure(Throwable t)
             MoreExecutors.directExecutor()
         );
       } else {
+        log.info("Set persist flag for the reason: %s.", String.join(",", persistReasons));
```

Comment:
Line 402 already has a similar log line. You just need to move it out of the `if`, i.e. move it to line 400.

Reply:
@kfaraz ping

Reply:
I think it makes sense to have the distinction. But the log line should reflect that this is the non-incremental code path.

```suggestion
        log.info("Marking ready for non-incremental async persist due to reasons[%s].", persistReasons);
```

Reply:
Updated the logging message to reflect this is non-incremental async path. 
@kfaraz